api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/request/MixedMetricsRequest.java

```diff
@@ -42,15 +42,27 @@
 
 
     public List<Gauge> getGaugeMetrics() {
-        return ImmutableList.copyOf(gaugeMetrics);
+        if(gaugeMetrics != null){
+            return ImmutableList.copyOf(gaugeMetrics);
+        }
+
+        return null;
     }
 
     public List<Availability> getAvailabilityMetrics() {
-        return ImmutableList.copyOf(availabilityMetrics);
+        if (availabilityMetrics != null) {
+            return ImmutableList.copyOf(availabilityMetrics);
+        }
+
+        return null;
     }
 
     public List<Counter> getCounters() {
-        return ImmutableList.copyOf(counters);
+        if (counters != null) {
+            return ImmutableList.copyOf(counters);
+        }
+
+        return null;
     }
 
```

Comment:
Wondering if we shouldn't simply drop the `ImmutableList.copyOf` and thus the new null check, as this a mutable object on purpose (representation of the mixed metrics entity) and not used anywhere else than in the handler code.